### PR TITLE
Related packages

### PR DESCRIPTION
**Pull request by JeffFessler**
Probably there are others that I don't know about but should...

### CODE REVIEW

**Comment by mortenpi** [comment]
Yes, I think it's a good idea to curate a list of related packages. I took the liberty of adding a bunch more.

File: README.md [hunk 1]
```diff
@@ -38,6 +38,26 @@ Usage questions can be posted on the [Julia Discourse forum][discourse-tag-url]
...
+## Related packages
+
+There are several packages that extend Documenter in different ways. The JuliaDocs organization maintains:
+
+* [DocumenterTools.jl](https://github.com/JuliaDocs/DocumenterTools.jl)
+* [DocumenterMarkdown.jl](https://github.com/JuliaDocs/DocumenterMarkdown.jl)
+
+Other third-party packages that can be combined with Documenter include:
+
+* [Literate.jl](https://github.com/fredrikekre/Literate.jl)
```
Comment (johnnychen94) [inline README.md:50]: alphabetic order maybe?

Comment (JeffFessler) [reply README.md:50]: Now that the maintainers have added so many other interesting packages I'll leave it to them to decide the order.

I was right that there were others that I didn't know about and am glad to learn about them here!

File: README.md [hunk 2]
```diff
@@ -38,6 +38,26 @@ Usage questions can be posted on the [Julia Discourse forum][discourse-tag-url]
...
+* [DemoCards.jl](https://github.com/johnnychen94/DemoCards.jl)
+* [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl)
+* [LiveServer.jl](https://github.com/tlienart/LiveServer.jl)
```
Comment (fredrikekre) [inline README.md:53]:
```suggestion
* [DemoCards.jl](https://github.com/johnnychen94/DemoCards.jl)
* [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl)
* [Literate.jl](https://github.com/fredrikekre/Literate.jl)
* [LiveServer.jl](https://github.com/tlienart/LiveServer.jl)
```

File: README.md [hunk 3]
```diff
@@ -38,6 +38,26 @@ Usage questions can be posted on the [Julia Discourse forum][discourse-tag-url]
...
+
+Finally, there are also a few other packages in the Julia ecosystem that are similar to Documenter, but fill a slightly different niche:
+
+* [Publish.jl](https://github.com/MichaelHatherly/Publish.jl)
+* [Franklin.jl](https://github.com/tlienart/Franklin.jl)
+* [Weave.jl](https://github.com/JunoLab/Weave.jl)
```
Comment (fredrikekre) [inline README.md:59]:
```suggestion
* [Franklin.jl](https://github.com/tlienart/Franklin.jl)
* [Publish.jl](https://github.com/MichaelHatherly/Publish.jl)
* [Weave.jl](https://github.com/JunoLab/Weave.jl)
```

**Comment by JeffFessler** [comment]
LGTM

**Pull request merged**